server/notebooks/api_views.py

```diff
@@ -31,34 +31,43 @@ def perform_destroy(self, instance):
             raise PermissionDenied
         super().perform_destroy(instance)
 
-    def perform_create(self, serializer):
-        with transaction.atomic():
-            if "owner" in self.request.data:
+    def create(self, request):
+        if "owner" in self.request.data:
+            if (
+                not self.request.data["owner"]
+                or self.request.data["owner"] == self.request.user.username
+            ):
+                owner = self.request.user
+            elif self.request.user.can_create_on_behalf_of_others:
                 User = get_user_model()
-                if (
-                    not self.request.data["owner"]
-                    or self.request.data["owner"] == self.request.user.username
-                ):
-                    owner = self.request.user
-                elif self.request.user.can_create_on_behalf_of_others:
-                    owner = get_object_or_404(User, username=self.request.data["owner"])
-                else:
-                    raise PermissionDenied
+                owner, _ = User.objects.get_or_create(username=self.request.data["owner"])
```

Comment:
99% sure it should be ok with the openidc middleware, but on testing this doesn't actually work properly with django social auth. Looks like we'll need to create a placeholder entry in `social_auth_usersocialauth` table.

We're actually not supporting this behaviour on the external server, but it's quite forseeable that we will need to (e.g. to support exporting a redash query to the public server). May as well get this right while this is fresh in our minds... will get to this soon.

Reply:
Ah, makes sense, thanks for elaborating. Looking through the Iodide code this felt like a missing part of the story so far, having a extensive user model and registration system -- probably because OIDC systems tend to ask applications to not implement so much of it. Which makes sense now, but I think it's worth keeping an eye on it while we figure out how Iodide.io will work in the future. E.g. having native support for other auth providers may make sense (e.g. Azure, GitLab, Bitbucket)